[stable/jenkins] Add Kubernetes plugin parameters

BryanHunt opened the pull request:
Signed-off-by: Bryan Hunt <bhunt@mac.com>

<!--
Thank you for contributing to helm/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/helm/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/helm/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the
history. This will make it easier to identify new changes. The PR will be squashed
anyways when it is merged. Thanks.

For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.

Please make sure you test your changes before you push them. Once pushed, a CircleCI
will run across your changes and do some initial checks and linting. These checks run
very quickly. Please check the results. We would like these checks to pass before we
even continue reviewing your changes.
-->
#### Is this a new chart

No

#### What this PR does / why we need it:

It add additional configuration parameters to the Kubernetes plugin

#### Which issue this PR fixes

  - fixes # 22212

#### Special notes for your reviewer:

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

BryanHuntNV committed:
Add Kubernetes plugin parameters

Signed-off-by: Bryan Hunt <bhunt@mac.com>
Signed-off-by: Bryan Hunt <bhunt@nvidia.com>

BryanHuntNV committed:
Add Kubernetes plugin parameters

Signed-off-by: Bryan Hunt <bhunt@mac.com>
Signed-off-by: Bryan Hunt <bhunt@nvidia.com>

wmcdona89 commented on stable/jenkins/README.md:
```diff
@@ -106,6 +106,11 @@ The following tables list the configurable parameters of the Jenkins chart and t
 | `master.slaveListenerPort`        | Listening port for agents            | `50000`                                   |
 | `master.slaveHostPort`            | Host port to listen for agents            | Not set                              |
 | `master.slaveKubernetesNamespace` | Namespace in which the Kubernetes agents should be launched  | Not set           |
+| `master.slaveDefaultsProviderTemplate` | The name of the pod template to use for providing default values | Not set  |
```
are you defining your agents in the chart via `agent` and `additionalAgents`?
all `additionalAgents` inherit from `agent` when the chart is deployed

that said, there can be value in setting Defaults Provider Template to ensure configuration changes made via the Jenkins console will be inherited as well

BryanHunt replied:
This is what I got from one of my co-workers:

> defaultProviderTemplate sets podTemplate() step to use the podTemplate we provide in the k8s cloud configuration. Without setting defaultProviderTemplate the template is just there but not used unless we do podTemplate(inheritFrom:'default)

wmcdona89 replied:
got it. you're only configuring `agent` in the chart and then defining pod templates in your pipelines which need to inherit from the pod template configured by `agent`. thus Defaults Provider Template needs to point to the pod template configured by `agent`.

wmcdona89 commented on stable/jenkins/README.md:
```diff
@@ -106,6 +106,11 @@ The following tables list the configurable parameters of the Jenkins chart and t
+| `master.slaveJenkinsUrl`          | Overrides the Kubernetes Jenkins URL    | Not set                                |
+| `master.slaveJenkinsTunnel`       | Overrides the Kubernetes Jenkins tunnel | Not set                                |
+| `master.slaveConnectTimeout`      | The connection timeout in seconds for connections to Kubernetes API. Minimum value is 5. | 5 |
+| `master.slaveReadTimeout`         | The read timeout in seconds for connections to Kubernetes API. Minimum value is 15. | 15 |
```
this does seem like the right place for all these values as they're used for the top-level configuration of the kubernetes plugin while `agent` values are used to configure pod templates. the `master.slave*` names could stand to be refactored down the road...perhaps `master.k8s.*`.

wmcdona89 commented on stable/jenkins/README.md:
```diff
@@ -223,6 +228,7 @@ Some third-party systems, e.g. GitHub, use HTML-formatted data in their payload
 | `agent.idleMinutes`        | Allows the Pod to remain active for reuse       | 0                      |
 | `agent.yamlTemplate`       | The raw yaml of a Pod API Object to merge into the agent spec | Not set  |
 | `agent.slaveConnectTimeout`| Timeout in seconds for an agent to be online    | 100                    |
+| `agent.jenkinsUrl`         | Overrides the value of the JENKINS_URL env      | Not set                |
```
could `master.slaveJenkinsUrl` be used to set JENKINS_URL or is JENKINS_URL always a unique value?

issue #22219 references JENKINS_URL

BryanHunt replied:
We are good with setting the environment variable the same as `master.slaveJenkinsUrl`.  Shall I make that change?

wmcdona89 replied:
yes, please make the change. I think it would be confusing to have both `master.slaveJenkinsUrl` and `agent.jenkinsUrl`.

wmcdona89 commented on stable/jenkins/values.yaml:
```diff
@@ -184,6 +184,8 @@ master:
   # slaveListenerLoadBalancerIP: 1.2.3.4
   slaveListenerServiceAnnotations: {}
+  slaveConnectTimeout: 5
+  slaveReadTimeout: 15
```
it seems like the defaults for `slaveConnectTimeout` and `slaveReadTimeout` should remain at 0

wmcdona89 replied:
please include the other new configurable fields here as well along with useful comments. fields that don't have default values may be commented out.

BryanHunt replied:
The documentation for the Kubernetes plugin states that the minimum value for `slaveConnectTimeout` and `slaveReadTimeout` is 5 and 15 respectively.  When using the UI to configure those values, you cannot enter values less that the minimum.

If you want the to default to 0, I'll change it, but that seems to be an invalid value.

wmcdona89 replied:
It appears the Kubernetes plugin behaves differently when the Configuration As Code plugin is installed. Without the Configuration As Code plugin, the values get set to the minimums even though the chart configures them as 0. With the Configuration As Code plugin, a value of 0 is allowed and means no timeout. Let's leave the chart defaults as 0.

wmcdona89 commented on stable/jenkins/templates/config.yaml:
```diff
@@ -45,6 +45,7 @@ data:
       <clouds>
         <org.csanchez.jenkins.plugins.kubernetes.KubernetesCloud plugin="kubernetes@{{ template "jenkins.kubernetes-version" . }}">
           <name>kubernetes</name>
+          <defaultsProviderTemplate>{{ default "" .Values.slaveDefaultsProviderTemplate }}</defaultsProviderTemplate>
```
please also add `defaultsProviderTemplate` to the JCasC [here](https://github.com/helm/charts/blob/master/stable/jenkins/templates/_helpers.tpl#L91)

wmcdona89 replied:
should be `.Values.master.slaveDefaultsProviderTemplate`

wmcdona89 commented on stable/jenkins/templates/config.yaml:
```diff
@@ -66,17 +67,29 @@ data:
-          <jenkinsUrl>http://{{ template "jenkins.fullname" . }}.{{ template "jenkins.namespace" . }}:{{.Values.master.servicePort}}{{ default "" .Values.master.jenkinsUriPrefix }}</jenkinsUrl>
-          <jenkinsTunnel>{{ template "jenkins.fullname" . }}-agent.{{ template "jenkins.namespace" . }}:{{ .Values.master.slaveListenerPort }}</jenkinsTunnel>
+{{- if .Values.master.slaveJenkinsUrl }}
+          <jenkinsUrl>{{ .Values.master.slaveJenkinsUrl }}</jenkinsUrl>
```
please also add `master.slaveJenkinsUrl` to the JCasC [here](https://github.com/helm/charts/blob/master/stable/jenkins/templates/_helpers.tpl#L93)

wmcdona89 commented on stable/jenkins/templates/config.yaml:
```diff
@@ -66,17 +67,29 @@ data:
+{{- end }}
+
+{{- if .Values.master.slaveJenkinsTunnel }}
+          <jenkinsTunnel>{{ .Values.master.slaveJenkinsTunnel }}</jenkinsTunnel>
```
please also add `master.slaveJenkinsTunnel` to the JCasC [here](https://github.com/helm/charts/blob/master/stable/jenkins/templates/_helpers.tpl#L92)

wmcdona89 reviewed (changes requested):
thanks for the PR!

BryanHuntNV committed:
Add changes to JCasC and changelog

Signed-off-by: Bryan Hunt <bhunt@nvidia.com>

BryanHuntNV committed:
Add changes to JCasC and changelog

Signed-off-by: Bryan Hunt <bhunt@nvidia.com>

wmcdona89 commented on stable/jenkins/templates/config.yaml:
```diff
@@ -66,17 +67,29 @@ data:
           <retentionTimeout>5</retentionTimeout>
-          <connectTimeout>0</connectTimeout>
-          <readTimeout>0</readTimeout>
+          <connectTimeout>{{ .Values.master.slaveConnectTimeout }}</connectTimeout>
```
please also add `master.slaveConnectTimeout` to the JCasC as `connectTimeout` [here](https://github.com/helm/charts/blob/master/stable/jenkins/templates/_helpers.tpl#L90)

wmcdona89 commented on stable/jenkins/templates/config.yaml:
```diff
@@ -66,17 +67,29 @@ data:
-          <connectTimeout>0</connectTimeout>
-          <readTimeout>0</readTimeout>
+          <connectTimeout>{{ .Values.master.slaveConnectTimeout }}</connectTimeout>
+          <readTimeout>{{ .Values.master.slaveReadTimeout }}</readTimeout>
```
please also add `master.slaveReadTimeout`to the JCasC as `readTimeout` [here](https://github.com/helm/charts/blob/master/stable/jenkins/templates/_helpers.tpl#L101)

wmcdona89 commented on stable/jenkins/templates/_helpers.tpl:
```diff
@@ -88,13 +88,26 @@ jenkins:
+      {{- if .Values.master.slaveJenkinsTunnel }}
+      jenkinsTunnel: "{{ .Values.master.slaveJenkinsTunnel }}"
+      {{- else }}
+      {{- if .Values.master.slaveKubernetesNamespace }}
```
let's make this an `else if`

wmcdona89 commented on stable/jenkins/templates/_helpers.tpl:
```diff
@@ -88,13 +88,26 @@ jenkins:
+      {{- if .Values.master.slaveJenkinsUrl }}
+      jenkinsUrl: "{{ .Values.master.slaveJenkinsUrl }}"
+      {{- else }}
       {{- if .Values.master.slaveKubernetesNamespace }}
```
let's make this an `else if`

wmcdona89 commented on stable/jenkins/templates/config.yaml:
```diff
@@ -66,17 +67,29 @@ data:
+{{- if .Values.master.slaveJenkinsTunnel }}
+          <jenkinsTunnel>{{ .Values.master.slaveJenkinsTunnel }}</jenkinsTunnel>
+{{- else }}
+  {{- if .Values.master.slaveKubernetesNamespace }}
```
let's make this an `else if`

wmcdona89 commented on stable/jenkins/templates/config.yaml:
```diff
@@ -66,17 +67,29 @@ data:
+{{- if .Values.master.slaveJenkinsUrl }}
+          <jenkinsUrl>{{ .Values.master.slaveJenkinsUrl }}</jenkinsUrl>
 {{- else }}
+  {{- if .Values.master.slaveKubernetesNamespace }}
```
let's make this an `else if`

BryanHuntNV committed:
Cleanup k8s changes

Signed-off-by: Bryan Hunt <bhunt@nvidia.com>

BryanHuntNV committed:
Cleanup k8s changes

Signed-off-by: Bryan Hunt <bhunt@nvidia.com>

BryanHunt committed:
Merge branch 'master' into kubernetes

BryanHuntNV committed:
Update chart version

Signed-off-by: Bryan Hunt <bhunt@nvidia.com>

BryanHuntNV committed:
Update chart version

Signed-off-by: Bryan Hunt <bhunt@nvidia.com>

BryanHuntNV committed:
Merge branch 'kubernetes' of https://github.com/BryanHunt/charts into kubernetes

gmichels committed:
[stable/selenium] Add liveness probes and preStop hooks on browsers (#22210)

* Add liveness probes and preStop hooks on browsers

Signed-off-by: Gustavo Michels <gustavomichels@icloud.com>

* Bump up chart version

Signed-off-by: Gustavo Michels <gustavomichels@icloud.com>

* Make liveness probes and preStop hooks optional and disabled by default

Signed-off-by: Gustavo Michels <gustavomichels@icloud.com>

* Update README to include new options

Signed-off-by: Gustavo Michels <gustavomichels@icloud.com>

yanjunli76 committed:
[stable/spark-history-server] add Custom S3 endpoints option (#22325)

Signed-off-by: Yanjun Li <yanjun.li76@hotmail.com>

Co-authored-by: Yanjun Li <yanjunli@vmware.com>

Stijn98s committed:
[stable/rocketchart] updated repository and tag (#22321)

* updated repository and tag

Signed-off-by: Stijn Smits <stijn@stijn98s.nl>

* revert repository url and updated to latest patch version

Signed-off-by: Stijn Smits <stijn@stijn98s.nl>

* updated chart versions

Signed-off-by: Stijn Smits <stijn@stijn98s.nl>

* update readme

Signed-off-by: Stijn Smits <stijn@stijn98s.nl>

mshalak-nix committed:
[stable/airflow] DAGs serialization without git-sync for web pods (#22335)

* Added parameter to switch on dag serialization without git-sync containers

Signed-off-by: Maksym Shalak <mshalak@n-ix.com>

* Added comment for serializeDAGs

Signed-off-by: Maksym Shalak <mshalak@n-ix.com>

* Added version

Signed-off-by: Maksym Shalak <mshalak@n-ix.com>

tuananhnguyen-ct committed:
upgrade to 2.18.1 (#22349)

Signed-off-by: Tuan Anh Nguyen <tuananh.nguyen-ext@commercetools.de>

yachtcaptain23 committed:
Upgrade metabase to 0.35.3 (#22320)

Signed-off-by: Albert Wang <amwang217@gmail.com>

mariusv committed:
[stable/kube2iam] bump kube2iam image and appVersion to v0.10.9 (#22333)

kube2iam v0.10.9 supports 1.17.3 due to client-go version upgrades

Signed-off-by: Marius Voila <myself@mariusv.com>

Commit:
[stable/chartmuseum] Update apiVersion for Kubernetes 1.16 support (#22315)

* Update apiVersion for Kubernetes 1.16 support

Signed-off-by: Alex Voloshyn <alex@geeoz.com>

* Add support for Kubernetes older than 1.16

Signed-off-by: Alex Voloshyn <alex@geeoz.com>

BryanHuntNV committed:
Missed a change to JENKINS_URL env

Signed-off-by: Bryan Hunt <bhunt@nvidia.com>

BryanHuntNV committed:
Missed a change to JENKINS_URL env

Signed-off-by: Bryan Hunt <bhunt@nvidia.com>

AmitThakkar committed:
#22269 [cluster-autoscaler] Adding cluster-autoscaler-priority-expander Config Map (#22302)

* #22269 [cluster-autoscaler] Adding cluster-autoscaler-priority-expander config map and parameters for supporting priority expander

Signed-off-by: AmitThakkar <vigildbest@gmail.com>

* #22269 [cluster-autoscaler] Fixed linting error.

Signed-off-by: AmitThakkar <vigildbest@gmail.com>

* Update stable/cluster-autoscaler/Chart.yaml

Signed-off-by: AmitThakkar <vigildbest@gmail.com>
Co-authored-by: Eric Bailey <yurrriq@users.noreply.github.com>

Co-authored-by: Eric Bailey <yurrriq@users.noreply.github.com>

wmcdona89 commented on stable/jenkins/CHANGELOG.md:
```diff
@@ -5,6 +5,10 @@ numbering uses [semantic versioning](http://semver.org).
+## 1.18.2
+
+Add support for executor mode configuraton with `master.executorMode`.
+
```
looks like you missed committing your change description

BryanHunt replied:
I used GitHub to resolve the merge conflict.  What do I need to do for this?

wmcdona89 replied:
this is currently just copy / paste of the verbiage from v1.18.1
it needs to be something like...

## 1.20.0

Add the following options for configuring the Kubernetes plugin.
- `master.slaveDefaultsProviderTemplate`
- `master.slaveJenkinsUrl`
- `master.slaveJenkinsTunnel`
- `master.slaveConnectTimeout`
-  `master.slaveReadTimeout`
```

wmcdona89 commented on stable/jenkins/values.yaml:
```diff
@@ -186,6 +186,9 @@ master:
   slaveListenerServiceAnnotations: {}
+  slaveDefaultsProviderTemplate: ""
+  slaveConnectTimeout: 0
+  slaveReadTimeout: 0
```
let's also add `master.slaveJenkinsUrl` and `master.slaveJenkinsTunnel` here so this file and the README are in sync

wmcdona89 replied:
oops...I discovered I was using an old version of the kubernetes plugin when looking into the minimum timeout values. looks like the minimums were established in [v1.21.5](https://github.com/jenkinsci/kubernetes-plugin/releases/tag/kubernetes-1.21.5)

so yeah...these should be 5 and 15 as you proposed

wmcdona89 commented on stable/jenkins/Chart.yaml:
```diff
@@ -1,7 +1,7 @@
 name: jenkins
 home: https://jenkins.io/
-version: 1.18.1
+version: 1.18.2
```
let's make this a minor version bump

tuananhnguyen-ct committed:
fix default non-ha state for alert-manager statefulset (#22350)

Signed-off-by: Tuan Anh Nguyen <tuananh.nguyen-ext@commercetools.de>

ErikLundJensen committed:
[Sealed secrets/optional] runAs and fsGroup have to be optional (#22369)

* #22367 stable/sealed-secrets RunAs in securityContext must be optional (OpenShift related)

Only set runAsUser and fsGroup if set.
Thereby it is possible to clear those values and deploy at OpenShift without any extra SCC changes.

Signed-off-by: Erik Lund Jensen <info@erikjensen.it>

* #22367 stable/sealed-secrets RunAs in securityContext must be optional

fixes #22367

Signed-off-by: Erik Lund Jensen <info@erikjensen.it>

* Bump chart version

fixes #22369

Signed-off-by: Erik Lund Jensen <info@erikjensen.it>

* Using fsGroup value from supplied values

fixes #22369

Signed-off-by: Erik Lund Jensen <info@erikjensen.it>

LukeChannings committed:
[stable/home-assistant]: ServiceMonitor bearerTokenSecret requires a SecretKeySelector (#22362)

* ServiceMonitor bearerTokenSecret requires a map

Prometheus ServiceMonitor Endpoint's `bearerTokenSecret` needs to be a [`SecretKeySelector`](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.17/#secretkeyselector-v1-core) not a string.

Ref: [Endpoint](https://github.com/coreos/prometheus-operator/blob/master/Documentation/api.md#endpoint)
Signed-off-by: Luke Channings <luke@channings.me>

* Fix version

Signed-off-by: Luke Channings <luke@channings.me>

BryanHuntNV committed:
Cleanup kubernetes configuration parameters
Plugin states min slaveConnectTimeout is 5
Plugin states min slaveReadTimeout is 15
Add slaveJenkinsUrl and slaveJenkinsTunnel to values.yaml

Signed-off-by: Bryan Hunt <bhunt@nvidia.com>

BryanHuntNV committed:
Cleanup kubernetes configuration parameters
Plugin states min slaveConnectTimeout is 5
Plugin states min slaveReadTimeout is 15
Add slaveJenkinsUrl and slaveJenkinsTunnel to values.yaml

Signed-off-by: Bryan Hunt <bhunt@nvidia.com>

L3n41c committed:
[stable/datadog] Update the example for `datadog.securityContext` (#22379)

Signed-off-by: Lénaïc Huard <lenaic.huard@datadoghq.com>

Commit:
[stable/openebs] Add the opportunity to change docker registry (#22372)

* Add the opportunity to change docker registry

Signed-off-by: olivier beyler <olivier.beyler@orange.com>

* Add parameter in README.md

Signed-off-by: olivier beyler <olivier.beyler@orange.com>

* fixup missing changement

Signed-off-by: olivier beyler <olivier.beyler@orange.com>

* fixup defaut value in readme

Signed-off-by: olivier beyler <olivier.beyler@orange.com>

davidkarlsen committed:
[stable/anchore-engine] use capabilities in order to support older in… (#22319)

* [stable/anchore-engine] use capabilities in order to support older ingress api

Signed-off-by: David Karlsen <david@davidkarlsen.com>

* fix chomping

Signed-off-by: David Karlsen <david@davidkarlsen.com>

tuananhnguyen-ct committed:
[stable/aws-iam-authenticator] Add apps/v1 support (#22326)

* add apps/v1 support

Signed-off-by: Tuan Anh Nguyen <tuananh.nguyen-ext@commercetools.de>

* add label selector

Signed-off-by: Tuan Anh Nguyen <tuananh.nguyen-ext@commercetools.de>

fabriziofortino committed:
[stable/cerebro] bump image (#22384)

Signed-off-by: fabriziofortino <fabrizio.fortino@gmail.com>

krmichel committed:
[stable/jenkins] Jenkins default configuration options (#22318)

* Add additional options to casc defaults

Signed-off-by: Kyle Michel <kylermichel@gmail.com>

* Bump chart version

Signed-off-by: Kyle Michel <kylermichel@gmail.com>

* Requested PR changes

Signed-off-by: Kyle Michel <kylermichel@gmail.com>

* Change how to override the markup formatter

Signed-off-by: Kyle Michel <kylermichel@gmail.com>

wmcdona89 commented on stable/jenkins/Chart.yaml:
```diff
@@ -1,7 +1,7 @@
 name: jenkins
 home: https://jenkins.io/
-version: 1.18.1
+version: 1.19.0
```
v1.19.0 was just merged. please rebase and bump to 1.20.0

iblackman committed:
Enable option to specify a different strategy than recreate (#22387)

Signed-off-by: Igor Valente Blackman <igor.blackman@gmail.com>

Laakso committed:
[stable/prometheus] Fix default non-HA state for alertmanager statefulset (#22386)

Since enableMeshPeer is false by default, cluster.listen-address should be empty.
Commit 5b4f507 caused regression and it is passing string as a parameter which is wrong. Documentation https://github.com/prometheus/alertmanager/blob/master/README.md#high-availability explicitly says cluster listen address should be defined like --cluster.listen-address= without passing emptry string.

Signed-off-by: Vesa Laakso <vesa.laakso@cfg.fi>

etiennetremel committed:
Add pod security policy support and create service account by default (#21796)

Signed-off-by: Etienne Tremel <etienne@kogusenn.com>

pwillie committed:
[stable/oauth2-proxy] Add podSecurityContext (#22181)

* [oauth2-proxy] Add podSecurityContext

Signed-off-by: Peter Wilson <dearpete@gmail.com>

* [oauth2-proxy] add podSecurityContext ci test

Signed-off-by: Peter Wilson <dearpete@gmail.com>

jasmdk committed:
Make web.allowedOrigins configurable using values file (#22224)

Signed-off-by: Jacob Stampe Mikkelsen <jmn@systematic.com>

LucasBoisserie committed:
Add vars for liveness and readyness (#22201)

Signed-off-by: LucasBoisserie <lucas.boisserie@gmail.com>

lawliet89 committed:
[stable/prometheus] Fix incorrect rendering of enableServiceLinks (#22391)

cf. https://github.com/helm/helm/issues/3308

Signed-off-by: Yong Wen Chua <lawliet89@users.noreply.github.com>

AmitThakkar committed:
[stable/autoscaler] scale-down-delay parameter name (#22396)

* [cluster-autoscaler] correcting scale-down-delay parameter.

Signed-off-by: AmitThakkar <vigildbest@gmail.com>

* [cluster-autoscaler] Chart Version bumped.

Signed-off-by: AmitThakkar <vigildbest@gmail.com>

BryanHuntNV committed:
Merge branch 'kubernetes' of https://github.com/BryanHunt/charts into kubernetes

Signed-off-by: Bryan Hunt <bhunt@nvidia.com>

BryanHuntNV committed:
Update chart version

Signed-off-by: Bryan Hunt <bhunt@nvidia.com>

wmcdona89 commented on stable/jenkins/templates/_helpers.tpl:
```diff
@@ -83,18 +83,29 @@ jenkins:
+      connectTimeout: "{{ .Values.master.slaveConnectTimeout }}"
+      readTimeout: "{{ .Values.master.slaveReadTimeout }}"
+      {{- if .Values.master.slaveJenkinsUrl }}
+      jenkinsUrl: "{{ .Values.master.slaveJenkinsUrl }}"
```
to allow templating this will be `jenkinsUrl: "{{ tpl .Values.master.slaveJenkinsUrl . }}"`

wmcdona89 commented on stable/jenkins/templates/_helpers.tpl:
```diff
@@ -83,18 +83,29 @@ jenkins:
       {{- end }}
+
+      {{- if .Values.master.slaveJenkinsTunnel }}
+      jenkinsTunnel: "{{ .Values.master.slaveJenkinsTunnel }}"
```
to allow templating this will be `jenkinsTunnel: "{{ tpl .Values.master.slaveJenkinsTunnel . }}"`

wmcdona89 commented on stable/jenkins/templates/_helpers.tpl:
```diff
@@ -269,7 +284,11 @@ Returns kubernetes pod template xml configuration
         <org.csanchez.jenkins.plugins.kubernetes.ContainerEnvVar>
           <key>JENKINS_URL</key>
+{{- if .Values.master.slaveJenkinsUrl }}
+          <value>{{ .Values.master.slaveJenkinsUrl }}</value>
```
to allow templating this will be `<value>{{ tpl .Values.master.slaveJenkinsUrl . }}</value>`

wmcdona89 commented on stable/jenkins/templates/config.yaml:
```diff
@@ -66,17 +67,25 @@ data:
           <namespace>{{ template "jenkins.master.slaveKubernetesNamespace" . }}</namespace>
-{{- if .Values.master.slaveKubernetesNamespace }}
+{{- if .Values.master.slaveJenkinsUrl }}
+          <jenkinsUrl>{{ .Values.master.slaveJenkinsUrl }}</jenkinsUrl>
```
to allow templating this will be `<jenkinsUrl>{{ tpl .Values.master.slaveJenkinsUrl . }}</jenkinsUrl>`

wmcdona89 commented on stable/jenkins/templates/config.yaml:
```diff
@@ -66,17 +67,25 @@ data:
+{{- end }}
+
+{{- if .Values.master.slaveJenkinsTunnel }}
+          <jenkinsTunnel>{{ .Values.master.slaveJenkinsTunnel }}</jenkinsTunnel>
```
to allow templating this will be `<jenkinsTunnel>{{ tpl .Values.master.slaveJenkinsTunnel . }}</jenkinsTunnel>`

wmcdona89 commented on stable/jenkins/templates/_helpers.tpl:
```diff
@@ -153,7 +164,11 @@ Returns kubernetes pod template configuration as code
     - containerEnvVar:
         key: "JENKINS_URL"
+    {{- if .Values.master.slaveJenkinsUrl }}
+        value: {{ .Values.master.slaveJenkinsUrl }}
```
to allow templating this will be `value: {{ tpl .Values.master.slaveJenkinsUrl . }}`

wmcdona89 reviewed (changes requested):
it just occurred to me that `master.slaveJenkinsUrl` and `master.slaveJenkinsTunnel` should allow templating. if you don't mind making the updates I'd appreciate it. I believe there are 6 spots to update.
